src/app/player/player.component.ts

```diff
@@ -124,6 +125,14 @@ export class PlayerComponent implements AfterViewInit, OnInit {
       this.timeLeft = `${hours ? hours : '00'}:${minutes}:${seconds}`;
     });
     this.api.getVideoDuration().subscribe((dur) => (this.duration = dur));
+
+    //convert seconds duration into hh:mm:ss format
+    let hours_tot = Math.floor(this.duration / 3600);
+    this.duration %= 3600;
+    let minutes_tot = Math.floor(this.duration / 60);
+    let seconds_tot = this.duration % 60;
+
+    this.durationTot = `${hours_tot ? hours_tot : '00'}:${minutes_tot}:${seconds_tot}`;
```

Comment:
Questo blocco di codice va bene ma va eseguito quando sappiamo che abbiamo a disposizione la durata (ce la manda il backend), quindi va messa dentro il subscribe di linea 127.
```suggestion
        this.api.getVideoDuration().subscribe((dur) => {
            this.duration = dur;
            
            // convert seconds duration into hh:mm:ss format
            let hours_tot = Math.floor(this.duration / 3600);
            this.duration %= 3600;
            let minutes_tot = Math.floor(this.duration / 60);
            let seconds_tot = this.duration % 60;
            this.durationTot = `${hours_tot ? hours_tot : '00'}:${minutes_tot}:${seconds_tot}`;
        });
```
Valuta anche se ha senso estrarre la logica di conversione ms in secondi in una funzione a parte, dato che la usiamo anche per calcolare il tempo rimanente.